KIPs/kip-dreaft_sdk_common_architecture.md

```diff
@@ -0,0 +1,928 @@
+
+An `Account` instance is injected into the `account` field of the AccountUpdate transaction types (`AccountUpdate`, `FeeDelegatedAccountUpdate` or `FeeDelegatedAccountUpdateWithRatio`). When an AccountUpdate transaction is successfully processed in Klaytn, the account key of the [Klaytn account] is updated with the `accountKey` defined in the `Account` instance.
+
+Interface for AccountKey is defined as `IAccountKey`, and account key classes (`AccountKeyLegacy`, `AccountKeyPublic`, `AccountKeyFail`, `AccountKeyWeightedMultiSig` and `AccountKeyRoleBased`) implements IAccountKey.
+
+`WeightedMultiSigOptions` is provided for `AccountKeyWeightedMultiSig` where the threshold and the weight of each key are defined.
+
+The `AccountKeyDecoder` class decodes the RLP-encoded string using the decode function implemented in each AccountKey class.
+
+#### IAccountKey
+
+| Method | Description |
+| ----------- | ----------- |
+| getRLPEncoding(): String | Returns a RLP-encoded string of AccountKey. AccountKey classes implement `IAccountKey`, and this function must be implemented. |
+
+#### AccountKeyLegacy
+
+| Method | Description |
+| ----------- | ----------- |
+| decode(rlpEncodedKey: String): AccountKeyLegacy | Decodes a RLP-encoded string of AccountKeyLegacy. |
+| getRLPEncoding(): String | Returns a RLP-encoded string of AccountKeyLegacy. |
+
+#### AccountKeyPublic
+
+| Method | Description |
+| ----------- | ----------- |
+| decode(rlpEncodedKey: String): AccountKeyPublic | Decodes RLP-encoded AccountKeyPublic. |
+| fromXYPoint(x: String, y: String): AccountKeyPublic | Creates an instance of AccountKeyPublic from x and y points. |
+| fromPublicKey(publicKey: String): AccountKeyPublic | Creates an instance of AccountKeyPublic from public key string. |
+| getXYPoint(): String[] | Returns the x and y points of the public key. |
+| getRLPEncoding(): String | Returns RLP-encoded string of AccountKeyPublic. |
+
+#### AccountKeyFail
+
+
+| Method | Description |
+| ----------- | ----------- |
+| decode(rlpEncodedKey: String): AccountKeyFail | Decodes RLP-encoded AccountKeyFail. |
+| getRLPEncoding(): String | Returns RLP-encoded string of AccountKeyFail. |
+
+#### AccountKeyWeightedMultiSig
+
+| Method | Description |
+| ----------- | ----------- |
+| decode(rlpEncodedKey: String): AccountKeyWeightedMultiSig | Decodes RLP-encoded AccountKeyWeightedMultiSig. |
+| fromPublicKeysAndOptions(pubArray: String[], options: WeightedMutliSigOptions): AccountKeyWeightedMultiSig | Creates an instance of AccountKeyWeightedMultiSig with public key strings and option that defines threshold and weight of each key. |
+| getRLPEncoding(): String | Returns RLP-encoded string of AccountKeyWeightedMultiSig. |
+
+#### WeightedPublicKey
+
+| Method | Description |
+| ----------- | ----------- |
+| encodeToBytes(): String[] | Returns an encoded weighted public key string. |
```

Comment:
```suggestion
| encodeToBytes(): String[] | Returns an encoded weighted public key string. |
```

What method to use for this encoding? RLP? or else?

Reply:
This function formats the weightedPublicKey for AccountKeyWeightedMultiSig RLP encoding. 
I think the name of this function should be modified. What do you think about modifying the name with `formatToEncode` or `formatForEncoding`?
It is a function implemented for internal use, so it doesn't matter if you change the name.
If you are concerned about the deployed already, how about leave the previously implemented function as it is and implement another function with the same name.

Reply:
I will remove this function from KIP :) Thank you for opinion.